mpqp/core/circuit.py

```diff
@@ -1134,12 +1134,46 @@ def to_other_language(
                 self.nb_qubits,
             )
         elif language == Language.CIRQ:
-            qasm2_code = self.to_other_language(Language.QASM2)
-            if TYPE_CHECKING:
-                assert isinstance(qasm2_code, str)
-            from mpqp.qasm.qasm_to_cirq import qasm2_to_cirq_Circuit
+            from cirq.circuits.circuit import Circuit as CirqCircuit
+            from cirq.ops.named_qubit import NamedQubit
+            from cirq.ops.identity import I
+
+            cirq_qubits = [NamedQubit(f"q_{i}") for i in range(self.nb_qubits)]
+            cirq_circuit = CirqCircuit()
+
+            for qubit in cirq_qubits:
+                cirq_circuit.append(I(qubit))
+
+            for instruction in self.instructions:
+                if isinstance(instruction, ExpectationMeasure):
+                    continue
+                elif isinstance(instruction, CustomGate):
+                    custom_circuit = QCircuit(self.nb_qubits)
+                    custom_circuit.add(instruction)
+                    qasm2_code = custom_circuit.to_other_language(Language.QASM2)
+                    if TYPE_CHECKING:
+                        assert isinstance(qasm2_code, str)
+                    from mpqp.qasm.qasm_to_cirq import qasm2_to_cirq_Circuit
+
+                    custom_cirq_circuit = qasm2_to_cirq_Circuit(qasm2_code)
+                    cirq_circuit += custom_cirq_circuit
```

Comment:
We are sure that the concatenation operation works well in for Cirq circuit ?

Reply:
The concatenation operation in Cirq works with the names of the qubits, and it functions correctly if we have the right name for each qubit. We need all the qubits, so we must apply the identity gate to each qubit.